src/util/strencodings.cpp

```diff
@@ -104,7 +106,7 @@ std::vector<unsigned char> ParseHex(const char* psz)
 
 std::vector<unsigned char> ParseHex(const std::string& str)
 {
-    return ParseHex(str.c_str());
+    return ParseHex(str.c_str(), str.size());
```

Comment:
3fad849f2f9d4c250e44fb7c329e1f93c14e48c7

NACK this change. Instead you can drop `ParseHex(const char*)` and move the code here with the proper changes - I've done it and `make check` ran fine.

Reply:
Well we have places where it's called with `const char*`. probably by dropping that it just calls the std::string constructor

Reply:
Same suggestion: can't we remove the `std::vector<unsigned char> ParseHex(const char* psz);` variant completely? 
Does this cause any actual issues if we did?
I'd like to move away from C-strings where possible and it seems more of a simplification.

Reply:
The only "problem" is a degregation in speed for places where we call this with C strings which I think is mostly(if not only?) tests(their raw strings usage)
It's probably worth the simplicity though.

Reply:
Is copying a string actually relevant performance wise here? Even if it was, parsing hex should only happen on user or rpc inputs, so I don't think we need to over-optimize this.

Reply:
@MarcoFalke in my benchmarks if you have `char*` and you convert it into `std::string` it spends almost the same time as the code before this PR (X2 slower). I think it's not about copying, it's about heap allocation. 
But if you prefer I'll be fine with removing the raw `char*` support because we probably only ever benefit from it in tests.

I agree that there's no need to over-optimize this but this is basically free optimization.

Reply:
Echoing the request of simply dropping `ParseHex(const char* psz)`.

Generally I think we should try move away from C-strings where possible.

Reply:
@elichai Is there a single call site that needs the raw string interface? If not, and you have performance concerns, you may explicitly delete the interface, and thus force all callers to explicitly call `ParseHex(std::string{psz})` and be aware of the performance impact.

Reply:
Checked now, it seems like `ParseHex(const char* psz)` is only used in tests except for 2 places:
https://github.com/bitcoin/bitcoin/blob/master/src/chainparams.cpp#L55
https://github.com/bitcoin/bitcoin/blob/master/src/net_processing.cpp#L2101

Both should almost never really run, so I'll just drop it